### PR TITLE
Adding example for drawstyle

### LINKED ISSUE #14724: drawstyle parameter of line needs example
`plot` takes a [drawstyle parameter](https://matplotlib.org/api/_as_gen/matplotlib.lines.Line2D.html?highlight=drawstyle#matplotlib.lines.Line2D.set_drawstyle) that can be used to modify how the points are connected. 
```python
fig, ax = plt.subplots(figsize=(15,5))
_ = ax.plot(x, sin_x, drawstyle='steps')
_ = ax.set_aspect('equal')
```
step

This isn't documented in the example gallery and it would be useful for someone to make an example in the same style as the [linestyle reference](https://matplotlib.org/gallery/lines_bars_and_markers/linestyles.html#sphx-glr-gallery-lines-bars-and-markers-linestyles-py)

### PR DESCRIPTION
**Pull request by Avni-Sharma**
Fixes #14724

Adds example for drawstyle parameter.

### CODE REVIEW

**Comment by Avni-Sharma** [comment]
cc @timhoffm

**Comment by story645** [comment]
Thanks for the PR! Can you please do what @timhoffm suggested in the issue and make this a section of the

**Comment by Avni-Sharma** [comment]
Hi @story645 . Exactly in which file(in the path) do I need to include that?

**Comment by timhoffm** [comment]
`examples/lines_bars_and_markers/step_demo.py`

File: examples/lines_bars_and_markers/step_demo.py [hunk 1]
```diff
@@ -29,6 +29,15 @@
 plt.legend(title='Parameter where:')
 plt.show()
 
+# The following example demonstrates the drawstyle parameter that can be used
```
Comment (timhoffm) [inline examples/lines_bars_and_markers/step_demo.py]: Please add a line of '#' to begin a new [sphinx gallery section](https://sphinx-gallery.github.io/syntax.html#embed-rst-in-your-example-python-files) and make this render as html, not as comment in the code. Compare also the references block bleow.

**Comment by tacaswell** [comment]
@Avni-Sharma Anything you need from us to move this PR forward?

**Comment by Avni-Sharma** [comment]
@tacaswell  I shall make the changes by EOD. Apologies for the delay

**Comment by timhoffm** [comment]
Since this did not move for some time and there were still some issues to be fixed, I decided to pick it up along with some other things in #15372. @Avni-Sharma thanks for your contribution!

**Pull request closed without merging**